src/language/languageServerController.ts

```diff
@@ -72,26 +72,48 @@ export default class LanguageServerController {
     );
   }
 
-  activate() {
+  async activate(): Promise<any> {
     // Start the client. This will also launch the server
     this.client.start();
-    this.client.onReady().then(() => {
-      /**
-       * TODO: Notification is for setup docs only.
-       */
-      this.client.onNotification('mongodbNotification', (messsage) => {
+
+    // Subscribe on notifications from the server when the client is ready
+    await this.client.onReady().then(() => {
+      this.client.onNotification('showInformationMessage', (messsage) => {
         vscode.window.showInformationMessage(messsage);
       });
+
+      this.client.onNotification('showErrorMessage', (messsage) => {
+        vscode.window.showErrorMessage(messsage);
+      });
     });
   }
 
-  deactivate(): Thenable<void> | undefined {
-    return this.client.stop();
+  deactivate(): void {
+    // Stop the language server
+    this.client.stop();
   }
 
   executeAll(codeToEvaluate: string, connectionString: string, connectionOptions: any = {}): Thenable<any> | undefined {
```

Comment:
Super off-topic but isn't `Thenable<any>` enough?

Reply:
Before refactoring, the `executeAll` function was returning the `this.client.sendRequest` method and `this.client` could be undefined. Now when I wrapped it into promise you are right, it returns always `Promise<any>` type. Nice catch! Will update it.